src/ws.ts

```diff
@@ -1,24 +1,26 @@
-import { createServer } from 'http';
+import { createServer, Server } from 'http';
 import {
   server as WebSocketServer,
   connection as Connection,
   request as Request,
   IMessage as Message
 } from 'websocket';
 
-import { fork, Sequence, Operation } from 'effection';
+import { fork, Operation } from 'effection';
 import { resumeOnCb } from './util';
 
 import { on } from '@effection/events';
 
-import { listen, ReadyCallback } from './http';
+import { listen } from './http';
 
-export function* createSocketServer(port: number, handler: ConnectionHandler, ready: ReadyCallback = x=>x): Sequence {
+interface UseServer { (server: Server): Operation }
+
+export function* createSocketServer(port: number, handler: ConnectionHandler, ready: UseServer): Operation {
   let server = createServer();
 
   yield listen(server, port);
 
-  ready(server);
+  yield ready(server);
```

Comment:
Since the callback needs to send `{connection: "ready"}` and `send` is an operation now, the callback has to also be an operation.